include/oneapi/dpl/internal/binary_search_impl.h

```diff
@@ -83,43 +83,49 @@ struct custom_brick
 template <class _Tag, typename Policy, typename InputIterator1, typename InputIterator2, typename OutputIterator,
```

Comment:
Overall question: why didn't we catch this earlier?

Reply:
We have catched this earlier and wrote issue https://github.com/oneapi-src/oneDPL/issues/1436

Reply:
I see. I would not say "it's not quite correct" but it doesn't match Tag Dispatching design. Thanks.

Reply:
Thanks, @rarutyun 
I fixed PR description.